metabase/metabase
# Add ability to turn a saved question into a dataset

The first frontend step on the data modeling project #18659 

This PR adds the ability to turn a saved question into a dataset and makes sure they're listed inside collections. Internally, datasets are saved questions with a `dataset` flag set to `true` (introduced in #18629 BE PR).

The implementation is as minimal as possible, so we can start playing around with it ASAP.

### To Verify

1. Open any saved question
2. Click the question name in the header
3. Click the new dataset icon in the left sidebar (aka the waffle icon 🧇 )
4. Click the "Turn this into a dataset" button in the education modal
5. If the question was not displayed as a table, it should be a table now
5. Open question's collection
6. Ensure it now has a purple data set icon instead of the previous one

### Demo

https://user-images.githubusercontent.com/17258145/139079970-58b6b7fb-a1db-4a66-99f1-6d0fd93f6df8.mov
